django/contrib/admin/options.py

```diff
@@ -1522,7 +1522,7 @@ def changelist_view(self, request, extra_context=None):
         if (request.method == "POST" and cl.list_editable and
                 '_save' in request.POST and not action_failed):
             FormSet = self.get_changelist_formset(request)
-            formset = cl.formset = FormSet(request.POST, request.FILES, queryset=cl.result_list)
+            formset = cl.formset = FormSet(request.POST, request.FILES, queryset=self.model._default_manager.get_queryset())
```

Comment:
It would probably be better to use ``self.get_queryset(request)`` instead to allow for better filtering, although I'm not sure that addresses your question.